### ISSUE TITLE
Confusing intermittent error when using `pytest-django` due to DB pollution between examples

### ISSUE THREAD

**Issue by knyghty** [HypothesisWorks/hypothesis#3216]
I'm experiencing some strange behaviour:

```python
class A(models.Model):
    name = models.TextField(unique=True)

class B(models.Model):
    a = models.ForeignKey(A, on_delete=models.PROTECT)


@pytest.mark.django_db
@given(b=from_model(B, a=from_model(A)))
def test_foo(b):
    assert b
```

This works fine.

However, adding a `TextField` to _B_ breaks it:

```python
class A(models.Model):
    name = models.TextField(unique=True)


class B(models.Model):
    a = models.ForeignKey(A, on_delete=models.PROTECT)
    name = models.TextField()


@pytest.mark.django_db
@given(b=from_model(B, a=from_model(A)))
def test_foo(b):
    assert b
```

This is broken with a `Flaky` error. The traceback is extremely long and doesn't seem very useful but I can post it if necessary.

It works fine, however, if you remove either the `TextField` on `B`, or the unique constraint on `A.name` - this also happens when using `UniqueConstratint` in the model's `Meta`.

It's a little peculiar -- I'm not sure what's causing it.

<details>
<summary>Traceback</summary>

```python-traceback
[gw1] linux -- Python 3.10.1 /usr/local/bin/python

self = <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>
sql = 'UPDATE "infrastructure_a" SET "name" = %s WHERE "infrastructure_a"."id" = %s'
params = ('0', 0)
ignored_wrapper_args = (False, {'connection': <django.db.backends.postgresql.base.DatabaseWrapper object at 0xffff9c51f760>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                # params default might be backend specific.
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)
E               psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "infrastructure_a_name_87b53ff0_uniq"
E               DETAIL:  Key (name)=(0) already exists.

/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py:85: UniqueViolation

The above exception was the direct cause of the following exception:

draw = <bound method ConjectureData.draw of ConjectureData(INTERESTING, 9 bytes, frozen)>
strat = builds(<django.db.models.manager.Manager object at 0xffff97244b80>.update_or_create, id=integers(-2147483648, 21474836...abet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))))

    @st.composite
    def _models_impl(draw, strat):
        """Handle the nasty part of drawing a value for models()"""
        try:
>           return draw(strat)[0]

/usr/local/lib/python3.10/site-packages/hypothesis/extra/django/_impl.py:139:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = builds(<django.db.models.manager.Manager object at 0xffff97244b80>.update_or_create, id=integers(-2147483648, 21474836...abet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))))
label = 6111981869288412008

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = builds(<django.db.models.manager.Manager object at 0xffff97244b80>.update_or_create, id=integers(-2147483648, 21474836...abet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
        try:
>           return self.target(
                *(data.draw(a) for a in self.args),
                **{k: data.draw(v) for k, v in self.kwargs.items()},
            )

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:777:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.models.manager.Manager object at 0xffff97244b80>, args = ()
kwargs = {'defaults': {'name': '0'}, 'id': 0}

    def manager_method(self, *args, **kwargs):
>       return getattr(self.get_queryset(), name)(*args, **kwargs)

/usr/local/lib/python3.10/site-packages/django/db/models/manager.py:85:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <QuerySet [<A: A object (-1447789849)>, <A: A object (110)>, <A: A object (62356)>, <A: A object (-9420)>, <A: A objec...39591430)>, <A: A object (-2682)>, <A: A object (75)>, <A: A object (-42463)>, '...(remaining elements truncated)...']>
defaults = {'name': '0'}, kwargs = {'id': 0}, obj = <A: A object (0)>
created = False, k = 'name', v = '0'

    def update_or_create(self, defaults=None, **kwargs):
        """
        Look up an object with the given kwargs, updating one with defaults
        if it exists, otherwise create a new one.
        Return a tuple (object, created), where created is a boolean
        specifying whether an object was created.
        """
        defaults = defaults or {}
        self._for_write = True
        with transaction.atomic(using=self.db):
            # Lock the row so that a concurrent update is blocked until
            # update_or_create() has performed its save.
            obj, created = self.select_for_update().get_or_create(defaults, **kwargs)
            if created:
                return obj, created
            for k, v in resolve_callables(defaults):
                setattr(obj, k, v)
>           obj.save(using=self.db)

/usr/local/lib/python3.10/site-packages/django/db/models/query.py:620:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <A: A object (0)>, force_insert = False, force_update = False
using = 'default', update_fields = None

    def save(self, force_insert=False, force_update=False, using=None,
             update_fields=None):
        """
        Save the current instance. Override this in a subclass if you want to
        control the saving process.

        The 'force_insert' and 'force_update' parameters can be used to insist
        that the "save" must be an SQL insert or update (or equivalent for
        non-SQL backends), respectively. Normally, they should not be set.
        """
        self._prepare_related_fields_for_save(operation_name='save')

        using = using or router.db_for_write(self.__class__, instance=self)
        if force_insert and (force_update or update_fields):
            raise ValueError("Cannot force both insert and updating in model saving.")

        deferred_fields = self.get_deferred_fields()
        if update_fields is not None:
            # If update_fields is empty, skip the save. We do also check for
            # no-op saves later on for inheritance cases. This bailout is
            # still needed for skipping signal sending.
            if not update_fields:
                return

            update_fields = frozenset(update_fields)
            field_names = set()

            for field in self._meta.concrete_fields:
                if not field.primary_key:
                    field_names.add(field.name)

                    if field.name != field.attname:
                        field_names.add(field.attname)

            non_model_fields = update_fields.difference(field_names)

            if non_model_fields:
                raise ValueError(
                    'The following fields do not exist in this model, are m2m '
                    'fields, or are non-concrete fields: %s'
                    % ', '.join(non_model_fields)
                )

        # If saving to the same database, and this model is deferred, then
        # automatically do an "update_fields" save on the loaded fields.
        elif not force_insert and deferred_fields and using == self._state.db:
            field_names = set()
            for field in self._meta.concrete_fields:
                if not field.primary_key and not hasattr(field, 'through'):
                    field_names.add(field.attname)
            loaded_fields = field_names.difference(deferred_fields)
            if loaded_fields:
                update_fields = frozenset(loaded_fields)

>       self.save_base(using=using, force_insert=force_insert,
                       force_update=force_update, update_fields=update_fields)

/usr/local/lib/python3.10/site-packages/django/db/models/base.py:743:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <A: A object (0)>, raw = False, force_insert = False, force_update = False
using = 'default', update_fields = None

    def save_base(self, raw=False, force_insert=False,
                  force_update=False, using=None, update_fields=None):
        """
        Handle the parts of saving which should be done only once per save,
        yet need to be done in raw saves, too. This includes some sanity
        checks and signal sending.

        The 'raw' argument is telling save_base not to save any parent
        models and not to do any changes to the values before save. This
        is used by fixture loading.
        """
        using = using or router.db_for_write(self.__class__, instance=self)
        assert not (force_insert and (force_update or update_fields))
        assert update_fields is None or update_fields
        cls = origin = self.__class__
        # Skip proxies, but keep the origin as the proxy model.
        if cls._meta.proxy:
            cls = cls._meta.concrete_model
        meta = cls._meta
        if not meta.auto_created:
            pre_save.send(
                sender=origin, instance=self, raw=raw, using=using,
                update_fields=update_fields,
            )
        # A transaction isn't needed if one query is issued.
        if meta.parents:
            context_manager = transaction.atomic(using=using, savepoint=False)
        else:
            context_manager = transaction.mark_for_rollback_on_error(using=using)
        with context_manager:
            parent_inserted = False
            if not raw:
                parent_inserted = self._save_parents(cls, using, update_fields)
>           updated = self._save_table(
                raw, cls, force_insert or parent_inserted,
                force_update, using, update_fields,
            )

/usr/local/lib/python3.10/site-packages/django/db/models/base.py:780:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <A: A object (0)>, raw = False, cls = <class 'ops.infrastructure.models.A'>
force_insert = False, force_update = False, using = 'default'
update_fields = None

    def _save_table(self, raw=False, cls=None, force_insert=False,
                    force_update=False, using=None, update_fields=None):
        """
        Do the heavy-lifting involved in saving. Update or insert the data
        for a single table.
        """
        meta = cls._meta
        non_pks = [f for f in meta.local_concrete_fields if not f.primary_key]

        if update_fields:
            non_pks = [f for f in non_pks
                       if f.name in update_fields or f.attname in update_fields]

        pk_val = self._get_pk_val(meta)
        if pk_val is None:
            pk_val = meta.pk.get_pk_value_on_save(self)
            setattr(self, meta.pk.attname, pk_val)
        pk_set = pk_val is not None
        if not pk_set and (force_update or update_fields):
            raise ValueError("Cannot force an update in save() with no primary key.")
        updated = False
        # Skip an UPDATE when adding an instance and primary key has a default.
        if (
            not raw and
            not force_insert and
            self._state.adding and
            meta.pk.default and
            meta.pk.default is not NOT_PROVIDED
        ):
            force_insert = True
        # If possible, try an UPDATE. If that doesn't update anything, do an INSERT.
        if pk_set and not force_insert:
            base_qs = cls._base_manager.using(using)
            values = [(f, None, (getattr(self, f.attname) if raw else f.pre_save(self, False)))
                      for f in non_pks]
            forced_update = update_fields or force_update
>           updated = self._do_update(base_qs, using, pk_val, values, update_fields,
                                      forced_update)

/usr/local/lib/python3.10/site-packages/django/db/models/base.py:862:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <A: A object (0)>
base_qs = <QuerySet [<A: A object (-1447789849)>, <A: A object (110)>, <A: A object (62356)>, <A: A object (-9420)>, <A: A objec...39591430)>, <A: A object (-2682)>, <A: A object (75)>, <A: A object (-42463)>, '...(remaining elements truncated)...']>
using = 'default', pk_val = 0
values = [(<django.db.models.fields.TextField: name>, None, '0')]
update_fields = None, forced_update = False

    def _do_update(self, base_qs, using, pk_val, values, update_fields, forced_update):
        """
        Try to update the model. Return True if the model was updated (if an
        update query was done and a matching row was found in the DB).
        """
        filtered = base_qs.filter(pk=pk_val)
        if not values:
            # We can end up here when saving a model in inheritance chain where
            # update_fields doesn't target any field in current model. In that
            # case we just say the update succeeded. Another case ending up here
            # is a model with just PK - in that case check that the PK still
            # exists.
            return update_fields is not None or filtered.exists()
        if self._meta.select_on_save and not forced_update:
            return (
                filtered.exists() and
                # It may happen that the object is deleted from the DB right after
                # this check, causing the subsequent UPDATE to return zero matching
                # rows. The same result can occur in some rare cases when the
                # database returns zero despite the UPDATE being executed
                # successfully (a row is matched and updated). In order to
                # distinguish these two cases, the object's existence in the
                # database is again checked for if the UPDATE query returns 0.
                (filtered._update(values) > 0 or filtered.exists())
            )
>       return filtered._update(values) > 0

/usr/local/lib/python3.10/site-packages/django/db/models/base.py:916:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <QuerySet [<A: A object (0)>]>
values = [(<django.db.models.fields.TextField: name>, None, '0')]

    def _update(self, values):
        """
        A version of update() that accepts field objects instead of field names.
        Used primarily for model saving and not intended for use by general
        code (it requires too much poking around at model internals to be
        useful at that level).
        """
        if self.query.is_sliced:
            raise TypeError('Cannot update a query once a slice has been taken.')
        query = self.query.chain(sql.UpdateQuery)
        query.add_update_fields(values)
        # Clear any annotations so that they won't be present in subqueries.
        query.annotations = {}
        self._result_cache = None
>       return query.get_compiler(self.db).execute_sql(CURSOR)

/usr/local/lib/python3.10/site-packages/django/db/models/query.py:809:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.models.sql.compiler.SQLUpdateCompiler object at 0xffff96f1d7b0>
result_type = 'cursor'

    def execute_sql(self, result_type):
        """
        Execute the specified update. Return the number of rows affected by
        the primary update query. The "primary update query" is the first
        non-empty query that is executed. Row counts for any subsequent,
        related queries are not available.
        """
>       cursor = super().execute_sql(result_type)

/usr/local/lib/python3.10/site-packages/django/db/models/sql/compiler.py:1591:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.models.sql.compiler.SQLUpdateCompiler object at 0xffff96f1d7b0>
result_type = 'cursor', chunked_fetch = False, chunk_size = 100

    def execute_sql(self, result_type=MULTI, chunked_fetch=False, chunk_size=GET_ITERATOR_CHUNK_SIZE):
        """
        Run the query against the database and return the result(s). The
        return value is a single data item if result_type is SINGLE, or an
        iterator over the results if the result_type is MULTI.

        result_type is either MULTI (use fetchmany() to retrieve all rows),
        SINGLE (only retrieve a single row), or None. In this last case, the
        cursor is returned if any query is executed, since it's used by
        subclasses such as InsertQuery). It's possible, however, that no query
        is needed, as the filters describe an empty set. In that case, None is
        returned, to avoid any unnecessary database interaction.
        """
        result_type = result_type or NO_RESULTS
        try:
            sql, params = self.as_sql()
            if not sql:
                raise EmptyResultSet
        except EmptyResultSet:
            if result_type == MULTI:
                return iter([])
            else:
                return
        if chunked_fetch:
            cursor = self.connection.chunked_cursor()
        else:
            cursor = self.connection.cursor()
        try:
>           cursor.execute(sql, params)

/usr/local/lib/python3.10/site-packages/django/db/models/sql/compiler.py:1202:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>
sql = 'UPDATE "infrastructure_a" SET "name" = %s WHERE "infrastructure_a"."id" = %s'
params = ('0', 0)

    def execute(self, sql, params=None):
>       return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)

/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>
sql = 'UPDATE "infrastructure_a" SET "name" = %s WHERE "infrastructure_a"."id" = %s'
params = ('0', 0), many = False
executor = <bound method CursorWrapper._execute of <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>>

    def _execute_with_wrappers(self, sql, params, many, executor):
        context = {'connection': self.db, 'cursor': self}
        for wrapper in reversed(self.db.execute_wrappers):
            executor = functools.partial(wrapper, executor)
>       return executor(sql, params, many, context)

/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>
sql = 'UPDATE "infrastructure_a" SET "name" = %s WHERE "infrastructure_a"."id" = %s'
params = ('0', 0)
ignored_wrapper_args = (False, {'connection': <django.db.backends.postgresql.base.DatabaseWrapper object at 0xffff9c51f760>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        self.db.validate_no_broken_transaction()
>       with self.db.wrap_database_errors:

/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py:80:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.utils.DatabaseErrorWrapper object at 0xffff973d2410>
exc_type = <class 'psycopg2.errors.UniqueViolation'>
exc_value = UniqueViolation('duplicate key value violates unique constraint "infrastructure_a_name_87b53ff0_uniq"\nDETAIL:  Key (name)=(0) already exists.\n')
traceback = <traceback object at 0xffff96f4e0c0>

    def __exit__(self, exc_type, exc_value, traceback):
        if exc_type is None:
            return
        for dj_exc_type in (
                DataError,
                OperationalError,
                IntegrityError,
                InternalError,
                ProgrammingError,
                NotSupportedError,
                DatabaseError,
                InterfaceError,
                Error,
        ):
            db_exc_type = getattr(self.wrapper.Database, dj_exc_type.__name__)
            if issubclass(exc_type, db_exc_type):
                dj_exc_value = dj_exc_type(*exc_value.args)
                # Only set the 'errors_occurred' flag for errors that may make
                # the connection unusable.
                if dj_exc_type not in (DataError, IntegrityError):
                    self.wrapper.errors_occurred = True
>               raise dj_exc_value.with_traceback(traceback) from exc_value

/usr/local/lib/python3.10/site-packages/django/db/utils.py:90:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>
sql = 'UPDATE "infrastructure_a" SET "name" = %s WHERE "infrastructure_a"."id" = %s'
params = ('0', 0)
ignored_wrapper_args = (False, {'connection': <django.db.backends.postgresql.base.DatabaseWrapper object at 0xffff9c51f760>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0xffff96f1d090>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                # params default might be backend specific.
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)
E               django.db.utils.IntegrityError: duplicate key value violates unique constraint "infrastructure_a_name_87b53ff0_uniq"
E               DETAIL:  Key (name)=(0) already exists.

/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py:85: IntegrityError

During handling of the above exception, another exception occurred:

self = FixedKeysDictStrategy(('b',), TupleStrategy((from_model(B, a=from_model(A)))))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data: ConjectureData) -> Ex:
        for _ in range(3):
            i = data.index
            try:
                data.start_example(MAPPED_SEARCH_STRATEGY_DO_DRAW_LABEL)
>               result = self.pack(data.draw(self.mapped_strategy))

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/strategies.py:818:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = TupleStrategy((from_model(B, a=from_model(A))))
label = 10510404995570551757

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = TupleStrategy((from_model(B, a=from_model(A))))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return tuple(data.draw(e) for e in self.element_strategies)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/collections.py:58:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <tuple_iterator object at 0xffff96f683a0>

>   return tuple(data.draw(e) for e in self.element_strategies)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/collections.py:58:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = from_model(B, a=from_model(A)), label = 15225241366338523813

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = from_model(B, a=from_model(A))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return data.draw(self.wrapped_strategy)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:164:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = _models_impl(builds(<bound method BaseManager._get_queryset_methods.<locals>.create_method.<locals>.manager_method of ...phabet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))
label = 15225241366338523813

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = _models_impl(builds(<bound method BaseManager._get_queryset_methods.<locals>.create_method.<locals>.manager_method of ...phabet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return data.draw(self.wrapped_strategy)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:164:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = <hypothesis.strategies._internal.core.CompositeStrategy object at 0xffff97247dc0>
label = 15225241366338523813

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <hypothesis.strategies._internal.core.CompositeStrategy object at 0xffff97247dc0>
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return self.definition(data.draw, *self.args, **self.kwargs)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1420:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

draw = <bound method ConjectureData.draw of ConjectureData(INTERESTING, 9 bytes, frozen)>
strat = builds(<django.db.models.manager.Manager object at 0xffff973a3340>.get_or_create, a=<hypothesis.strategies._internal.c... object at 0xffff97247d00>, name=text(characters(blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>))

    @st.composite
    def _models_impl(draw, strat):
        """Handle the nasty part of drawing a value for models()"""
        try:
>           return draw(strat)[0]

/usr/local/lib/python3.10/site-packages/hypothesis/extra/django/_impl.py:139:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = builds(<django.db.models.manager.Manager object at 0xffff973a3340>.get_or_create, a=<hypothesis.strategies._internal.c... object at 0xffff97247d00>, name=text(characters(blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>))
label = 6111981869288412008

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = builds(<django.db.models.manager.Manager object at 0xffff973a3340>.get_or_create, a=<hypothesis.strategies._internal.c... object at 0xffff97247d00>, name=text(characters(blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
        try:
            return self.target(
                *(data.draw(a) for a in self.args),
>               **{k: data.draw(v) for k, v in self.kwargs.items()},
            )

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:779:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <dict_itemiterator object at 0xffff96f61f80>

>       **{k: data.draw(v) for k, v in self.kwargs.items()},
    )

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:779:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = <hypothesis.strategies._internal.core.CompositeStrategy object at 0xffff97247d00>
label = 15225241366338523813

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <hypothesis.strategies._internal.core.CompositeStrategy object at 0xffff97247d00>
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return self.definition(data.draw, *self.args, **self.kwargs)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1420:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

draw = <bound method ConjectureData.draw of ConjectureData(INTERESTING, 9 bytes, frozen)>
strat = builds(<django.db.models.manager.Manager object at 0xffff97244b80>.update_or_create, id=integers(-2147483648, 21474836...abet=characters(blacklist_categories=('Cs',), blacklist_characters='\x00'), min_size=1).filter(lambda s: <unknown>)))))

    @st.composite
    def _models_impl(draw, strat):
        """Handle the nasty part of drawing a value for models()"""
        try:
            return draw(strat)[0]
        except IntegrityError:
>           reject()

/usr/local/lib/python3.10/site-packages/hypothesis/extra/django/_impl.py:141:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def reject() -> NoReturn:
>       raise UnsatisfiedAssumption()
E       hypothesis.errors.UnsatisfiedAssumption

/usr/local/lib/python3.10/site-packages/hypothesis/control.py:24: UnsatisfiedAssumption

During handling of the above exception, another exception occurred:

self = <hypothesis.core.StateForActualGivenExecution object at 0xffff97207160>
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def _execute_once_for_engine(self, data):
        """Wrapper around ``execute_once`` that intercepts test failure
        exceptions and single-test control exceptions, and turns them into
        appropriate method calls to `data` instead.

        This allows the engine to assume that any exception other than
        ``StopTest`` must be a fatal error, and should stop the entire engine.
        """
        try:
            trace = frozenset()
            if (
                self.failed_normally
                and not self.failed_due_to_deadline
                and Phase.shrink in self.settings.phases
                and Phase.explain in self.settings.phases
                and sys.gettrace() is None
                and not PYPY
            ):  # pragma: no cover
                # This is in fact covered by our *non-coverage* tests, but due to the
                # settrace() contention *not* by our coverage tests.  Ah well.
                tracer = Tracer()
                try:
                    sys.settrace(tracer.trace)
                    result = self.execute_once(data)
                    if data.status == Status.VALID:
                        self.explain_traces[None].add(frozenset(tracer.branches))
                finally:
                    sys.settrace(None)
                    trace = frozenset(tracer.branches)
            else:
>               result = self.execute_once(data)

/usr/local/lib/python3.10/site-packages/hypothesis/core.py:699:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <hypothesis.core.StateForActualGivenExecution object at 0xffff97207160>
data = ConjectureData(INTERESTING, 9 bytes, frozen), print_example = False
is_final = False, expected_failure = None

    def execute_once(
        self, data, print_example=False, is_final=False, expected_failure=None
    ):
        """Run the test function once, using ``data`` as input.

        If the test raises an exception, it will propagate through to the
        caller of this method. Depending on its type, this could represent
        an ordinary test failure, or a fatal error, or a control exception.

        If this method returns normally, the test might have passed, or
        it might have placed ``data`` in an unsuccessful state and then
        swallowed the corresponding control exception.
        """

        data.is_find = self.is_find

        text_repr = [None]
        if self.settings.deadline is None:
            test = self.test
        else:

            @proxies(self.test)
            def test(*args, **kwargs):
                self.__test_runtime = None
                initial_draws = len(data.draw_times)
                start = time.perf_counter()
                result = self.test(*args, **kwargs)
                finish = time.perf_counter()
                internal_draw_time = sum(data.draw_times[initial_draws:])
                runtime = datetime.timedelta(
                    seconds=finish - start - internal_draw_time
                )
                self.__test_runtime = runtime
                current_deadline = self.settings.deadline
                if not is_final:
                    current_deadline = (current_deadline // 4) * 5
                if runtime >= current_deadline:
                    raise DeadlineExceeded(runtime, self.settings.deadline)
                return result

        def run(data):
            # Set up dynamic context needed by a single test run.
            with local_settings(self.settings):
                with deterministic_PRNG():
                    with BuildContext(data, is_final=is_final):

                        # Generate all arguments to the test function.
                        args, kwargs = data.draw(self.search_strategy)
                        if expected_failure is not None:
                            text_repr[0] = arg_string(test, args, kwargs)

                        if print_example or current_verbosity() >= Verbosity.verbose:
                            output = StringIO()

                            printer = RepresentationPrinter(output)
                            if print_example:
                                printer.text("Falsifying example:")
                            else:
                                printer.text("Trying example:")

                            if self.print_given_args:
                                printer.text(" ")
                                printer.text(test.__name__)
                                with printer.group(indent=4, open="(", close=""):
                                    printer.break_()
                                    for v in args:
                                        printer.pretty(v)
                                        # We add a comma unconditionally because
                                        # generated arguments will always be kwargs,
                                        # so there will always be more to come.
                                        printer.text(",")
                                        printer.breakable()

                                    for i, (k, v) in enumerate(kwargs.items()):
                                        printer.text(k)
                                        printer.text("=")
                                        printer.pretty(v)
                                        printer.text(",")
                                        if i + 1 < len(kwargs):
                                            printer.breakable()
                                printer.break_()
                                printer.text(")")
                            printer.flush()
                            report(output.getvalue())
                        return test(*args, **kwargs)

        # Run the test function once, via the executor hook.
        # In most cases this will delegate straight to `run(data)`.
>       result = self.test_runner(data, run)

/usr/local/lib/python3.10/site-packages/hypothesis/core.py:637:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = ConjectureData(INTERESTING, 9 bytes, frozen)
function = <function StateForActualGivenExecution.execute_once.<locals>.run at 0xffff96f26320>

    def default_new_style_executor(data, function):
>       return function(data)

/usr/local/lib/python3.10/site-packages/hypothesis/executors.py:47:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def run(data):
        # Set up dynamic context needed by a single test run.
        with local_settings(self.settings):
            with deterministic_PRNG():
                with BuildContext(data, is_final=is_final):

                    # Generate all arguments to the test function.
>                   args, kwargs = data.draw(self.search_strategy)

/usr/local/lib/python3.10/site-packages/hypothesis/core.py:596:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = TupleStrategy((just(()), fixed_dictionaries({'b': from_model(B, a=from_model(A))}).map(lambda args: dict(args, **kwargs))))
label = 9447048757421901795

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
                return strategy.do_draw(self)
            else:
                strategy.validate()
                try:
>                   return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:878:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = TupleStrategy((just(()), fixed_dictionaries({'b': from_model(B, a=from_model(A))}).map(lambda args: dict(args, **kwargs))))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return tuple(data.draw(e) for e in self.element_strategies)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/collections.py:58:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <tuple_iterator object at 0xffff96ee58d0>

>   return tuple(data.draw(e) for e in self.element_strategies)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/collections.py:58:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = fixed_dictionaries({'b': from_model(B, a=from_model(A))}).map(lambda args: dict(args, **kwargs))
label = 10700208128082112041

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = fixed_dictionaries({'b': from_model(B, a=from_model(A))}).map(lambda args: dict(args, **kwargs))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data: ConjectureData) -> Ex:
        for _ in range(3):
            i = data.index
            try:
                data.start_example(MAPPED_SEARCH_STRATEGY_DO_DRAW_LABEL)
>               result = self.pack(data.draw(self.mapped_strategy))

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/strategies.py:818:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = fixed_dictionaries({'b': from_model(B, a=from_model(A))})
label = 12000696373305623442

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = fixed_dictionaries({'b': from_model(B, a=from_model(A))})
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data):
>       return data.draw(self.wrapped_strategy)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:164:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen)
strategy = FixedKeysDictStrategy(('b',), TupleStrategy((from_model(B, a=from_model(A)))))
label = 12000696373305623442

    def draw(self, strategy, label=None):
        if self.is_find and not strategy.supports_find:
            raise InvalidArgument(
                f"Cannot use strategy {strategy!r} within a call to find "
                "(presumably because it would be invalid after the call had ended)."
            )

        at_top_level = self.depth == 0
        if at_top_level:
            # We start this timer early, because accessing attributes on a LazyStrategy
            # can be almost arbitrarily slow.  In cases like characters() and text()
            # where we cache something expensive, this led to Flaky deadline errors!
            # See https://github.com/HypothesisWorks/hypothesis/issues/2108
            start_time = time.perf_counter()

        strategy.validate()

        if strategy.is_empty:
            self.mark_invalid()

        if self.depth >= MAX_DEPTH:
            self.mark_invalid()

        if label is None:
            label = strategy.label
        self.start_example(label=label)
        try:
            if not at_top_level:
>               return strategy.do_draw(self)

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = FixedKeysDictStrategy(('b',), TupleStrategy((from_model(B, a=from_model(A)))))
data = ConjectureData(INTERESTING, 9 bytes, frozen)

    def do_draw(self, data: ConjectureData) -> Ex:
        for _ in range(3):
            i = data.index
            try:
                data.start_example(MAPPED_SEARCH_STRATEGY_DO_DRAW_LABEL)
                result = self.pack(data.draw(self.mapped_strategy))
                data.stop_example()
                return result
            except UnsatisfiedAssumption:
>               data.stop_example(discard=True)

/usr/local/lib/python3.10/site-packages/hypothesis/strategies/_internal/strategies.py:822:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ConjectureData(INTERESTING, 9 bytes, frozen), discard = True

    def stop_example(self, discard=False):
        if self.frozen:
            return
        if discard:
            self.has_discards = True
        self.depth -= 1
        assert self.depth >= -1
        self.__example_record.stop_example(discard)

        labels_for_structure = self.labels_for_structure_stack.pop()

        if not discard:
            if self.labels_for_structure_stack:
                self.labels_for_structure_stack[-1].update(labels_for_structure)
            else:
                self.tags.update([structural_coverage(l) for l in labels_for_structure])

        if discard:
            # Once we've discarded an example, every test case starting with
            # this prefix contains discards. We prune the tree at that point so
            # as to avoid future test cases bothering with this region, on the
            # assumption that some example that you could have used instead
            # there would *not* trigger the discard. This greatly speeds up
            # test case generation in some cases, because it allows us to
            # ignore large swathes of the search space that are effectively
            # redundant.
            #
            # A scenario that can cause us problems but which we deliberately
            # have decided not to support is that if there are side effects
            # during data generation then you may end up with a scenario where
            # every good test case generates a discard because the discarded
            # section sets up important things for later. This is not terribly
            # likely and all that you see in this case is some degradation in
            # quality of testing, so we don't worry about it.
            #
            # Note that killing the branch does *not* mean we will never
            # explore below this point, and in particular we may do so during
            # shrinking. Any explicit request for a data object that starts
            # with the branch here will work just fine, but novel prefix
            # generation will avoid it, and we can use it to detect when we
            # have explored the entire tree (up to redundancy).

>           self.observer.kill_branch()

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:940:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <hypothesis.internal.conjecture.datatree.TreeRecordingObserver object at 0xffff96f57dc0>

    def kill_branch(self):
        """Mark this part of the tree as not worth re-exploring."""
        if self.killed:
            return

        self.killed = True

        if self.__index_in_current_node < len(self.__current_node.values) or (
            self.__current_node.transition is not None
            and not isinstance(self.__current_node.transition, Killed)
        ):
>           inconsistent_generation()

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/datatree.py:375:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def inconsistent_generation():
>       raise Flaky(
            "Inconsistent data generation! Data generation behaved differently "
            "between different runs. Is your data generation depending on external "
            "state?"
        )
E       hypothesis.errors.Flaky: Inconsistent data generation! Data generation behaved differently between different runs. Is your data generation depending on external state?

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/datatree.py:29: Flaky

During handling of the above exception, another exception occurred:

>   ???

ops/infrastructure/tests.py:48:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def inconsistent_generation():
>       raise Flaky(
            "Inconsistent data generation! Data generation behaved differently "
            "between different runs. Is your data generation depending on external "
            "state?"
        )
E       hypothesis.errors.Flaky: Inconsistent data generation! Data generation behaved differently between different runs. Is your data generation depending on external state?

/usr/local/lib/python3.10/site-packages/hypothesis/internal/conjecture/datatree.py:29: Flaky
```
</details>

**Comment by Zac-HD** [comment]
It's useful to include the traceback even if it's long and you've given a reproducing script (:heart_eyes:), so that I can check that we see the *same* error.  Happily, GitHub will automatically put a scrollbar on long code blocks, and you can even enclose that in an HTML `<details>` block if you want a single-line collapsed view:

```
    <details>
    <summary>Click here to see a long traceback</summary>

    ```python-traceback
    # paste your traceback here
    ```
    </details>
```

More directly, this sure sounds like it might be a weird bug we should fix; I'm moving at the moment but should be able to investigate within a few weeks.

**Comment by knyghty** [comment]
Thanks, I added the TB 👍🏻

**Comment by Zac-HD** [comment]
Ah, it looks like you're not resetting the database between examples - maybe you're [using `TestCase` when you need `TransactionTestCase`](https://hypothesis.readthedocs.io/en/latest/django.html#hypothesis.extra.django.TestCase)?

**Comment by knyghty** [comment]
Ah, I (think I) see my problem. It's not that I need `TransactionTestCase`, it's that I need to use the _hypothesis_ `TestCase`. since I was just using pytest test functions, it wasn't using any `TestCase` class, Django's hypothesis', or otherwise, so I get the "wrong" error.

So unless I'm missing something there's no support for using pytest with Django without also using the hypothesis `TestCase` classes? That's reasonable, I can mix and match, there is even some benefit in doing so - but I suppose I'd like to reframe this issue then, as a request to add support for using pytest-django directly, and not needing to use the `TestCase` classes, if that's possible.

**Comment by Zac-HD** [comment]
Ah, yep, that's it then!  

- I wonder if there's a way we can auto-detect this, and emit a warning which explains what you need to do?
- See #2382: we'd love to have compatibility with `pytest-django`, it's just that nobody on the Hypothesis team has the required expertise at the moment.  
  (are you our lucky developer??? :star_struck::wink:)

**Comment by Zac-HD** [comment]
Closing this in favor of https://github.com/pytest-dev/pytest-django/issues/912, since I think `pytest-django` is a better place to solve the problem 🙂
